GccBoardProject1/src/loraDriver.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * loraDriver.h
+ *
+ * Created: 3/24/2017 12:08:35 PM
+ *  Author: mazacar8
+ */ 
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#ifndef LORADRIVER_H_
+#define LORADRIVER_H_
+
+#define MAX_JOIN_ATTEMPTS 5
+#define MAX_STATUS_CHECKS 5
+#define LORA_BAUD 57600
+#define DEBUG_BAUD 115200
+#define RX_BUFFER_SIZE 100
+#define TX_BUFFER_SIZE 100
+
+struct port_config reset_config;
+struct usart_module usart_instance;
+struct usart_module lora_uart_instance;
+
+volatile uint8_t rx_buffer[RX_BUFFER_SIZE];
+volatile uint8_t tx_buffer[TX_BUFFER_SIZE];
+
+void lora_setup(void);
+void lora_join(void);
```

Comment:
Void returns for most of these functions are fine, but since join has a variety of outcomes this should probably communicate some more information back to the calling function. You could define a `lora_error_t` struct or similar, or just change it to an int and `#define` some return values.